Dockerfile

```diff
@@ -106,6 +107,46 @@ VOLUME /var/lib/zoneminder/images /var/lib/zoneminder/events /var/lib/mysql /var
 # To speed up configuration testing, we put it here
 ADD utils/docker /ZoneMinder/utils/docker/
 
-CMD /ZoneMinder/utils/docker/setup.sh && /ZoneMinder/utils/docker/start.sh >/var/log/start.log 2>&1 & /bin/bash
-
-# Run example docker run -it -p 1080:80 -e PHP_TIMEZONE='Europe/Paris' -v /disk/zoneminder/events:/var/lib/zoneminder/events -v /disk/zoneminder/images:/var/lib/zoneminder/images -v /disk/zoneminder/mysql:/var/lib/mysql -v /disk/zoneminder/logs:/var/log/zm --name zoneminder zoneminder/zoneminder
+CMD if [ ! -z "$TZ" ]; then \
+			echo "date.timezone= $TZ" >> /etc/php/7.0/apache2/php.ini; \
+			else \
+			echo "date.timezone= America/Los_Angeles" >> /etc/php/7.0/apache2/php.ini; \
+	fi && \
+  if [ ! -z "$MYSQL_SERVER" && ! -z "$MYSQL_USER" &&  ! -z "$MYSQL_PASSWORD" && ! -z "$MYSQL_DB"]; then \
+		sed -i -e "s/ZM_DB_NAME=zm/ZM_DB_NAME=$MYSQL_USER/g" /etc/zm.conf && \
+		sed -i -e "s/ZM_DB_USER=zmuser/ZM_DB_USER=$MYSQL_USER/g" /etc/zm.conf && \
+		sed -i -e "s/ZM_DB_PASS=zm/ZM_DB_PASS=$MYSQL_PASS/g" /etc/zm.conf && \
+		sed -i -e "s/ZM_DB_HOST=localhost/ZM_DB_HOST=$MYSQL_SERVER/g" /etc/zm.conf; \
+	else \
+	  usermod -d /var/lib/mysql/ mysql && \
+		service mysql restart && \
+		mysql -u root -e "create database zm;" && \
+		mysql -u root -e "GRANT ALL PRIVILEGES ON *.* TO 'zmuser'@'localhost' IDENTIFIED BY 'zmpass';"; \
+		mysql -u root zm < /usr/local/share/zoneminder/db/zm_create.sql; \
+	fi && \
+	/ZoneMinder/utils/docker/setup.sh && \
+	service apache2 restart && \
+	/ZoneMinder/utils/docker/start.sh >/var/log/start.log 2>&1
+
+# Run example if you don't have seperate db:
+# docker run -d -t -p 1080:80 \
+#    -e PHP_TIMEZONE='America/Los_Angeles' \
+#    -v /disk/zoneminder/events:/var/lib/zoneminder/events \
+#    -v /disk/zoneminder/images:/var/lib/zoneminder/images \
+#    -v /disk/zoneminder/mysql:/var/lib/mysql \
+#    -v /disk/zoneminder/logs:/var/log/zm \
+#    --name zoneminder \
+#    zoneminder/zoneminder
+
+# Run example if you have a seperate db:
+# docker run -d -t -p 1080:80 \
+#     -e PHP_TIMEZONE='America/Los_Angeles' \
+#     -e ZM_DB_NAME='zmuser' \
```

Comment:
Don't mind me, but this should be `ZM_DB_USER`

Thanks for putting this together.

Reply:
Thanks!  I'll make sure to get that in there.  It also points out that my env var for timezones (in the example) is incorrect as well.

Reply:
Actually looking at this... and how those variables are used... Doesnt this only work when building the dockerfile? These variables get used when building around line 115... Does the CMD command get rerun so we are able to update the /etc/zm.conf? It seems that we would need to use some kind of other startup script that would modify that file on each startup based on env variables... Unless I have just completely overlooked something.